libbeat/common/dtfmt/dtfmt_test.go

```diff
@@ -0,0 +1,87 @@
+package dtfmt
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFormat(t *testing.T) {
+	tests := []struct {
+		time     time.Time
+		pattern  string
+		expected string
+	}{
+		// year.month.day of month
+		{mkDate(6, 8, 1), "y.M.d", "6.8.1"},
+		{mkDate(2006, 8, 1), "y.M.d", "2006.8.1"},
+		{mkDate(2006, 8, 1), "yy.MM.dd", "06.08.01"},
+		{mkDate(6, 8, 1), "yy.MM.dd", "06.08.01"},
+		{mkDate(2006, 8, 1), "yyy.MMM.dd", "2006.Aug.01"},
+		{mkDate(2006, 8, 1), "yyyy.MMMM.d", "2006.August.1"},
+		{mkDate(2006, 8, 1), "yyyyyy.MM.ddd", "002006.08.001"},
+
+		// year of era.month.day
+		{mkDate(6, 8, 1), "Y.M.d", "6.8.1"},
+		{mkDate(2006, 8, 1), "Y.M.d", "2006.8.1"},
+		{mkDate(2006, 8, 1), "YY.MM.dd", "06.08.01"},
+		{mkDate(6, 8, 1), "YY.MM.dd", "06.08.01"},
+		{mkDate(2006, 8, 1), "YYY.MMM.dd", "2006.Aug.01"},
+		{mkDate(2006, 8, 1), "YYYY.MMMM.d", "2006.August.1"},
+		{mkDate(2006, 8, 1), "YYYYYY.MM.ddd", "002006.08.001"},
+
+		// week year + week of year + day of week
+		{mkDate(2015, 1, 1), "xx.ww.e", "15.01.4"},
+		{mkDate(2014, 12, 31), "xx.ww.e", "15.01.3"},
+		{mkDate(2015, 1, 1), "xx.w.E", "15.1.Thu"},
+		{mkDate(2014, 12, 31), "xx.w.E", "15.1.Wed"},
+		{mkDate(2015, 1, 1), "xx.w.EEEE", "15.1.Thursday"},
+		{mkDate(2014, 12, 31), "xx.w.EEEE", "15.1.Wednesday"},
```

Comment:
Maybe also add a test with `xxxx.ww` which is the format recommended by the Logstash docs.